eventauth/eventcontent.go

```diff
@@ -0,0 +1,280 @@
+package eventauth
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
+// creaateContent is the JSON content of a m.room.create event along with
+// the top level keys needed for auth.
+type createContent struct {
+	// We need the domain of the create event when checking federatebility.
+	senderDomain string
+	// We need the roomID to check that events are in the same room as the create event.
+	roomID string
```

Comment:
I thought so, and yeah it's an interesting one. I think it can be confusing to pollute namespaces like this, but I understand that it may be even worse to have to pass around the same 2 structs to everything. Let's leave it for now and see how bad this gets as you add more logic.